src/data_provider/src/osinfo/sysOsInfoWin.cpp

```diff
@@ -24,12 +24,12 @@ static std::string getVersion(const bool isMinor = false)
     if(IsWindowsVistaOrGreater())
     {
         Utils::Registry currentVersion{HKEY_LOCAL_MACHINE, R"(SOFTWARE\Microsoft\Windows NT\CurrentVersion)"};
-        if (IsWindows8OrGreater())
+        try
         {
             const auto versionNumber{currentVersion.dword(isMinor ? "CurrentMinorVersionNumber" : "CurrentMajorVersionNumber")};
```

Comment:
We should use the no throw version of Registry::dword and check the returned bool.